### ISSUE TITLE
copying a artifact with ':' in filename fails

### ISSUE THREAD

**Issue by nacc** [devopshq/artifactory#245]
I'm trying to copy a artifact from one repository to another on the same instance. It has a ':' in its name

`3:6.04~git20190206.bf6db5b4+dfsg1-2`

It would appear that when I do a artifact lookup and then do

`artifact.copy(...)`, the `:` does not get encoded and I get:

`
Failed to copy from src=SRCREPO/pool/internal/focal/extlinux_3:6.04~git20190206.bf6db5b4+dfsg1-2_amd64.deb to target=TGTREPO/pool/internal/focal/extlinux_3%3a6.04~git20190206.bf6db5b4+dfsg1-2_amd64.deb: Could not find item at SRCREPOl:pool/internal/focal/extlinux_3:6.04~git20190206.bf6db5b4+dfsg1-2_amd64.deb`

The source artifact is available as

`extlinux_3%3a6....`

**Comment by beliaev-maksim** [comment]
can you please try my global fix and let me know if that works for your case: #247

**Comment by nacc** [comment]
@beliaev-maksim I hit this exact issue with the same change locally. So you have to (afaict) take apart the pathobj into its parts to make sure the drive part does not get encoded (since it has `http://` in it or so) but the filenames do.

**Comment by beliaev-maksim** [comment]
@nacc 
yes, but the problem is that if we encode `ArtifactoryPath`, then for example `path.name` will return you `my%3Afile` instead of `my:file`

**Comment by nacc** [comment]
Yep, so I think you can only encode it just before interacting with the REST API (that is, when transforming from a path to a URL). It probably deserves a helper method.

**Comment by beliaev-maksim** [comment]
@allburov please close as addressed by #254

**Comment by beliaev-maksim** [comment]
addressed by #254
